packages/terra-date-picker/src/DateInput.jsx

```diff
@@ -592,11 +566,11 @@ const DatePickerInput = (props) => {
       return;
     }
     // increment current valid date by 1 day, if not valid date set date to tomorrow instead
-    if (event.key === '=' || event.key === '+') {
+    if (event.key === '=' || event.key === '+' || event.keyCode === KEY_UP) {
       if (validDate) {
-        inputDate = DateUtil.incrementDateByDay(inputDate, DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        inputDate = DateUtil.incrementDateByDay(inputDate, DateUtil.ISO_EXTENDED_DATE_FORMAT, inputTypeValue);
       } else {
-        inputDate = DateUtil.incrementDateByDay(DateUtil.getCurrentDate(), DateUtil.ISO_EXTENDED_DATE_FORMAT);
+        inputDate = DateUtil.incrementDateByDay(DateUtil.getCurrentDate(), DateUtil.ISO_EXTENDED_DATE_FORMAT, inputTypeValue);
```

Comment:
Lets rename this function to `incrementDate`

Reply:
Updated - https://github.com/cerner/terra-framework/pull/1999/commits/7b61de3934e318f97325758b279207800dc84d9e